File: src/main/java/editor/util/StringUtils.java
package editor.util;

/**
 * This interface contains some useful string utilities to make formatting easier.
 * 
 * @author Alec Roelke
 */
public interface StringUtils
{
    /**
     * Format a double with the desired precision as if with %f, unless it's a whole number,
     * in which case forgo the decimal place.
     * 
     * @param n number to format
     * @param precision precision to use if the number is fractional
     * @return The formatted string.
     */
    static String formatDouble(double n, int precision)
    {
        if (n == (int)n)
            return Integer.toString((int)n);
        else
            return String.format("%." + precision + "f", n);
    }    
}

File: src/main/scala/editor/util/StringUtils.scala
package editor.util

/**
 * Utilities for formatting strings.
 * @author Alec Roelke
 */
object StringUtils {
  /**
   * Format a number into a string with no decimal places if it's an integer or with a cutomizable precision otherwise.
   * 
   * @param n number to format
   * @param precision precision to use to format it if it isn't an integer
   * @return a string containing the formatted number
   */
  def formatDouble(n: Double, precision: Int) = {
    if (n == n.toInt)
      n.toInt.toString
    else
      s"%.${precision}f".format(n)
  }
}
